### PR TITLE
Fixed issue where attachments < 3mb were not being encoded correctly

### PR DESCRIPTION
**Pull request by mrashed-dev**
# Description
This PR fixes an issue where we were not encoding attachments less than 3mb properly. Furthermore, for `CreateAttachmentRequest` you may now provide your own base64 encoded string if you prefer to not have the SDK encode it.

# License
I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: src/resources/messages.ts [hunk 1]
```diff
@@ -210,7 +210,18 @@ export class Messages extends Resource {
     if (attachmentSize >= Messages.MAXIMUM_JSON_ATTACHMENT_SIZE) {
       requestOptions.form = Messages._buildFormRequest(requestBody);
     } else {
-      requestOptions.body = requestBody;
+      if (requestBody.attachments) {
```
Comment (VictorMonteiro7) [inline src/resources/messages.ts:213]: As the documentation said:
_`To add attachments to a draft with a total HTTP payload size under 3MB, you can use either the application/json format or the multipart/form-data format.`_

The reduce above only gets the last attachment's size. 
I think the correct reduce would be:
```ts
 const maxLength = attachments.reduce((acc, attachment) => {
   acc += (attachment.size as number) || 0;
     return acc;
    }, 0) || 0;
```

Comment (mrashed-dev) [reply src/resources/messages.ts:213]: 🤦🏼 Of course. Thanks for pointing this out, will rectify it 😄

File: src/resources/drafts.ts [hunk 2]
```diff
@@ -131,6 +132,15 @@ export class Drafts extends Resource {
         form,
```
Comment (yifanplanet) [inline src/resources/drafts.ts:132]: Same comment goes for this:
```suggestion
  const attachmentSize =
      requestBody.attachments?.reduce((total, attachment) => {
        return total + (attachment.size || 0);
      }, 0) || 0;
```

File: src/resources/drafts.ts [hunk 3]
```diff
@@ -167,6 +177,15 @@ export class Drafts extends Resource {
...
```
Comment (yifanplanet) [inline src/resources/drafts.ts:177]: And this:
```
  const attachmentSize =
    requestBody.attachments?.reduce((total, attachment) => {
      return total + (attachment.size || 0);
  }, 0) || 0;
```

**Review by yifanplanet — APPROVED** [review]
Just those two changes needed for the src/resources/drafts.ts

**Pull request merged**